src/WorkerChannel.ts

```diff
@@ -304,4 +333,18 @@ export class WorkerChannel implements IWorkerChannel {
 
     return status;
   }
+
+  private runInvocationRequestBefore(context: Context, userFunction: Function): Function {
+    let wrappedFunction = userFunction;
+    for (let before of this._invocationRequestBefore) {
+      wrappedFunction = before(context, wrappedFunction);

```

Comment:
Just double checking that you don't want to make anything async? Could do a Promise.all

Reply:
I could see these being useful as Promises. Currently each `before` updates `wrappedFunction` on completion. Is there a nice way to accomplish this with Promise.all?

Reply:
Thoughts on await on `before` and `after` calls? Otherwise this PR looks good!